dot/types/babe.go

```diff
@@ -107,7 +110,7 @@ func GetSlotFromHeader(header *Header) (uint64, error) {
 
 	preDigest, ok := header.Digest.Types[0].Value().(PreRuntimeDigest)
 	if !ok {
-		return 0, fmt.Errorf("first digest item is not pre-digest")
+		return 0, ErrNoFirstPreDigest
```

Comment:
Would be nice to put in the message the type we got
```suggestion
		return 0, fmt.Errorf("%w: got %T", ErrNoFirstPreDigest, header.Digest.Types[0].Value())
```